### PR TITLE
Allow multiple CIDR arguments to run/start/attach/detach

### PR DESCRIPTION
**Pull request by awh**
Addresses #467.

### CODE REVIEW

**Comment by rade** [comment]
docs need updating

File: weave [hunk 1]
```diff
@@ -641,16 +653,20 @@ case "$COMMAND" in
...
+            CIDRS="$CIDRS $1"
+            shift 1
+        done
         CONTAINER=$(docker run $DNS_ARGS -d "$@")
         if [ "$LOCAL" = "--local" ]; then
             create_bridge
-            with_container_netns $CONTAINER attach $CIDR >/dev/null
-            tell_dns PUT $CONTAINER $CIDR
+            with_container_netns $CONTAINER attach $CIDRS >/dev/null
+            for CIDR in $CIDRS; do
```
Comment (awh) [inline weave]: I will refactor `tell_dns` to take multiple CIDRs and hence eliminate the loop here and elsewhere...

File: weave [hunk 2]
```diff
@@ -641,16 +653,20 @@ case "$COMMAND" in
...
             [ "$1" = "--with-dns" ] && shift 1
         fi
-        CIDR=$1
-        shift 1
-        validate_cidr $CIDR
+        CIDRS=""
+        while is_cidr "$1"; do
+            CIDRS="$CIDRS $1"
+            shift 1
+        done
```
Comment (awh) [inline weave]: This `while` over `is_cidr` appears in four places, but I'm struggling to see how to factor it out as it modifies `$@` with `shift`. Any thoughts @dpw?

Comment (dpw) [inline weave]: shift $(collect_cidr_args "$@")

`collect_cidr_args` counts the CIDR parameters and sets `CIDRS` as a side effect (Matthias will love that).

Comment (awh) [inline weave]: Just spent twenty minutes banging my head against this, wondering why it wasn't working - unfortunately the `$()` expression executes in a subshell, so the modified global `$CIDRS` is not visible to the calling process...

**Comment by awh** [comment]
Updated documentation and factored out some common parts. Unfortunately the CIDR collection loop is still resisting (see my comment elsewhere) - remaining avenues are the CPS transform that @dpw mentioned or a function that exports *two* global variables. Thoughts @rade?

File: site/features.md [hunk 3]
```diff
@@ -90,6 +90,14 @@ each other but not the containers of our first application...
 This isolation-through-subnets scheme is an example of carrying over a
 well-known technique from the 'on metal' days to containers.
 
+If desired, a container can be attached to multiple subnets when it is
+started:
+
+    host1# weave run 10.2.2.1/24 10.2.3.1/24 -t -i ubuntu
+
+In this case, all of the addresses are added to the container's
+weave interface before it is brought up.
```
Comment (rade) [inline site/features.md]: Drop that last sentence; it's an implementation detail. And also wrong ;) (the interface is called 'ethwe').

**Comment by rade** [comment]
Two globals is fine. No need for the 'GLOBAL_' prefix; all our vars are global.

**Comment by awh** [comment]
Docs updated, CIDR parsing refactored.

**Comment by rade** [comment]
Any reason not do do 'expose' and 'hide' too?

**Comment by awh** [comment]
`weave expose` and `weave hide` now accept multiple addresses.

**Comment by awh** [comment]
Docs updated to use consistent subnets in examples.

**Comment by awh** [comment]
I'm going to have a stab at rebasing this to clean up the history - please hold off merging for now.

**Comment by rade** [comment]
I really don't care. But if you must...

**Pull request merged**